near-plugins/tests/contracts/access_controllable/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+use near_plugins::{access_control, AccessControlRole, AccessControllable};
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{env, log, near_bindgen, AccountId};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+#[derive(AccessControlRole, BorshSerialize, BorshDeserialize)]
+pub enum Role {
+    Level1,
+    Level2,
+    Level3,
+}
+
+#[access_control(storage_prefix = "__foo", role_type = "Role")]
+#[near_bindgen]
+#[derive(Default, BorshDeserialize, BorshSerialize)]
+pub struct StatusMessage {
+    records: HashMap<AccountId, String>,
+}
+
+#[near_bindgen]
+impl StatusMessage {
+    #[payable]
+    pub fn set_status(&mut self, message: String) {
+        let account_id = env::signer_account_id();
+        log!("{} set_status with message {}", account_id, message);
+        self.records.insert(account_id, message);
+    }
+
+    pub fn get_status(&self, account_id: AccountId) -> Option<String> {
+        log!("get_status for account_id {}", account_id);
+        self.records.get(&account_id).cloned()
+    }
+
+    // The contract can interact with Acl by:
+    //
+    // a) Calling functions, e.g.
+    //    self.acl.has_role(role, account_id)
+    //
+    // b) Using attributes (not yet implemented), e.g.
+    //    #[acl_any(Role::Level1, Role::Level2)]
+    //    pub fn foo(&mut self) {}
```

Comment:
Once this function is implemented, recommended tests:

- Account with no role should fail
- Account with role level 1 should succeed
- Account with role level 2 should succeed
- Account with role level 3 should fail
- Account with roles level 1 and level 3 should succeed
- Account with role level 1 Admin should fail
- Account with role super admin should fail

Reply:
The first 5 cases are covered in this PR. Admin related functionality will be added in a follow up PR. There's a TODO in the code to then add the last two cases.